explainer.md

```diff
@@ -395,13 +395,143 @@ function checkMagicWindowSupport() {
 }
 ```
 
+## Input
+
+XR hardware provides a wide variety of input mechanisms, ranging from single state buttons to fully tracked controllers with multiple buttons, joysticks, triggers, or touchpads. While the intent is to eventually support the full range of available hardware, for the initial version of the WebXR Device API the focus is on enabling a more universal "point and click" style system that can be supported in some capacity by any known XR device and in magic window mode.
+
+In this model every input source has a ray that indicates what is being pointed at, called the "Pointer Ray", and reports when the primary action for that device has been triggered, surfaced as a "select" event. When the select event is fired the XR application can use the pointer ray of the input source that generated the event to determine what the user was pointing at and respond accordingly.
+
+### Enumerating input sources
+
+Calling the `getInputSources()` function on an `XRSession` will return a list of all currently available `XRInputSource`s. An `XRInputSource` may represent a tracked controller, inputs built into the headset itself, or more ephemeral input mechanims like tracking of hand gestures. When input sources are added to or removed from the list of available input sources the `inputsourceschange` event will be fired on the `XRSession` object to indicate that any cached copies of the list should be refreshed.
+
+```js
+// Get the current list of input sources.
+let xrInputSources = xrSession.getInputSources();
+
+// Update the list of input sources if it ever changes.
+xrSession.addEventListener('inputsourceschange', (ev) => {
+  xrInputSources = xrSession.getInputSources();
+});
+```
+
+### Input poses
+
+Each input source can queried a `XRInputPose` using the `getInputPose()` function of any `XRPresentationFrame`. Getting the pose requires passing in the `XRInputSource` you want the pose for, as well as the `XRFrameOfReference` the pose values should be given in, just like `getDevicePose()`. Similar to `getDevicePose()` the requested pose may return `null` in cases where tracking has been lost.
```

Comment:
getInputPose (and any other APIs that give back poses) should just take an XRCoordinateSystem, to give apps the most flexibility around which coordinate system they reason within.

Reply:
That actually is what it takes in, according to the IDL at the bottom of the doc. I've fixed the reference here.